index.js

```diff
@@ -55,9 +57,22 @@ module.exports = (secret) => (socket, next) => {
         }
       }
       return handler.call(this, ...args);
-    });
+    };
+    
+    handlers.set(handler, newHandler);
+    return socket[on](event, newHandler);
   };
 
+  socket.off = (event, handler) => {
+    if (reservedEvents.includes(event)) return socket[off](event, handler);
+
+    if (handlers.has(handler)) {
```

Comment:
I think it should also remove the handler from the Map, to prevent memory leak, no?